src/k8s/pkg/utils/file.go

```diff
@@ -76,6 +76,30 @@ func GetServiceArgument(service string, argument string) (string, error) {
 	return matches[1], nil
 }
 
+// UpdateServiceArgs updates the value of an argument in a service arguments file.
+func UpdateServiceArgs(argument, value, service string) error {
+	argument = "--" + argument
+	configFile := SnapDataPath("args", service)
+	replaceLine := argument + "=" + value
+
+	fileContent, err := os.ReadFile(configFile)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return os.WriteFile(configFile, []byte(replaceLine+"\n"), 0o644)
+		}
+		return fmt.Errorf("failed to read %s config file: %w", service, err)
+	}
+
+	regexPattern := regexp.MustCompile(`(?m)^` + regexp.QuoteMeta(argument) + `=.*$`)
+	if regexPattern.Match(fileContent) {
+		fileContent = regexPattern.ReplaceAll(fileContent, []byte(replaceLine))
+	} else {
+		fileContent = append(fileContent, []byte(replaceLine+"\n")...)
+	}
+
+	return os.WriteFile(configFile, fileContent, 0o644)
+}
```

Comment:
please see https://github.com/canonical/k8s-snap/pull/49/files#diff-506efc1ed4c8530427a69550ef2bbc3caba657b9a33dde1851cc3b1f07ecf17aR39 (there was also some discussion around why want to avoid a regex for updating arguments in the same PR).

My preference would be to wait until #49 is merged and adjust the code. We need to move #49 forward.

Reply:
I see, I'll wait until #49 is merged to refactor the PR. Thanks for the heads up :)